libraries/AP_Quicktune/AP_Quicktune.cpp

```diff
@@ -0,0 +1,672 @@
+#include "AP_Quicktune.h"
+
+#if QUICKTUNE_ENABLED
+
+const AP_Param::GroupInfo AP_Quicktune::var_info[] = {
+    // @Param: ENABLE
+    // @DisplayName: Quicktune enable
+    // @Description: Enable quicktune system
+    // @Values: 0:Disabled,1:Enabled
+    // @User: Standard
+    AP_GROUPINFO_FLAGS("ENABLE", 1, AP_Quicktune, enable, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: AXES
+    // @DisplayName: Quicktune axes
+    // @Description: Axes to tune
+    // @Bitmask: 0:Roll,1:Pitch,2:Yaw
+    // @User: Standard
+    AP_GROUPINFO("AXES", 2, AP_Quicktune, axes_enabled, 7),
+
+    // @Param: DOUBLE_TIME
+    // @DisplayName: Quicktune doubling time
+    // @Description: Time to double a tuning parameter. Raise this for a slower tune.
+    // @Range: 5 20
+    // @Units: s
+    // @User: Standard
+    AP_GROUPINFO("DOUBLE_TIME", 3, AP_Quicktune, double_time, 10),
+
+    // @Param: GAIN_MARGIN
+    // @DisplayName: Quicktune gain margin
+    // @Description: Reduction in gain after oscillation detected. Raise this number to get a more conservative tune
+    // @Range: 20 80
+    // @Units: %
+    // @User: Standard
+    AP_GROUPINFO("GAIN_MARGIN", 4, AP_Quicktune, gain_margin, 60),
```

Comment:
This is not gain margin so the name is inappropriate. This is a backoff from where oscillation is detected (this can also be triggered by noise).

This algorithm increases the gains until oscillation is strong enough to be detected. This is requires a loop gain of greater than 1 (commonly expressed as a unit factor or dB but never %). So the loop gain will be some factor over 1, it is hard to know how much higher than 1 but it could easily be 2. The parameter range above suggests reducing the gain by a factor of 0.8 to 0.2 from where oscillation is detected. A reduction of only 20% is a gain margin of less than 0.2 (2dB) but could be negative. Normal gain margins are between 1/0.5–1/0.4,  2–2.5, 6-8dB.

The only reason to use a value of above 50% in the parameter above is if the oscillation detection is triggering off noise, not oscillation. 

Percentage is not a unit used in control systems and appears unprofessional, it also requires us to carry a multiplication by 100 around. dB is probably most common engineering term used but is both difficult for the lay person and requires an even more complex calculation. 
A simple factor is:
1. an appropriate engineering term (looks professional),
2. easy for the lay person to understand,
3. does not require any calculations to be done to use in the code.

We have established a pattern for this in the PID loops to allow plane to scale parameters differently to copter:

```
// set slew limiter scale factor
void set_slew_limit_scale(int8_t scale) { _slew_limit_scale = scale; }
```

Reply:
> Percentage is not a unit used in control systems and appears unprofessional

this really bugs me. percentages are used all over the place in engineering. When you first taught me to manually tune you told me to backoff by a percentage (typically 30 to 50 percent). Go and look at airbus or boeing engineering documents, they are full of percentages.

Proportions in control maths makes sense. Parameters are not maths. Parameters are user interfaces. User interfaces in aviation usually use percentages not proportions. 

We have 72 percentage parameters in ArduPilot. Only some are in plane. Many in heli.

Reply:
Just because we use something regularly doesn't mean it is appropriate in all circumstances. Using it for Gain Margin is one example that is clearly inappropriate.

Using percentage to express reducing a parameter is an example that would be appropriate. However, my other arguments still apply in that case.

A simple factor is:
1. an appropriate engineering term (looks professional),
2. easy for the lay person to understand,
3. does not require any calculations to be done to use in the code.

I have chosen to use the unit factors throughout the control code because it is most universally appropriate in an engineering context, easy to understand and maintains consistency and clarity. It would be nice if new PR's acknowledged that effort and attempted to maintain it.

Reply:
what is appropriate for APIs is not the same as what is appropriate for user facing parameters. Users are not all engineers, and even within engineering percentages are extremely common, just look around the cockpit of an airliner